include/boost/beast/experimental/core/url_parser.hpp

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_EXPERIMENTAL_CORE_URL_PARSER_HPP
+#define BOOST_BEAST_EXPERIMENTAL_CORE_URL_PARSER_HPP
+
+#include <boost/beast/core/string.hpp>
+#include <cctype> // for std::isalpha etc...
+
+namespace boost {
+namespace beast {
+
+    //--------------------------------------------------------------------------
+    // scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]
+
+    using boost::beast::string_view;
+
+    struct url_parser
```

Comment:
This needs a Javadoc comment

Reply:
Will add one. New to exemplar comments, I'll do my best.

Reply:
The bar for stuff in experimental/ is lower so you could at a minimum get away with just one line:
```
/// Parses URLs
```
Or
```
/** Parses URLs
*/
```
Eventually you will want to add something to the quick reference:
https://github.com/boostorg/beast/blob/develop/doc/qbk/quickref.xml#L295

No one has ever thought to update the reference but hey, first time for everything I guess! If you don't do it, I will, so no big deal (and setting up the toolchain for building the docs is kind of a hassle so I will understand).